programs/address-lookup-table/src/processor.rs

```diff
@@ -151,6 +194,83 @@ impl Processor {
         Ok(())
     }
 
+    fn create_lookup_table_v2(
+        invoke_context: &mut InvokeContext,
+        untrusted_recent_slot: Slot,
+        bump_seed: u8,
+    ) -> Result<(), InstructionError> {
+        let transaction_context = &invoke_context.transaction_context;
+        let instruction_context = transaction_context.get_current_instruction_context()?;
+
+        let lookup_table_account =
+            instruction_context.try_borrow_instruction_account(transaction_context, 0)?;
+        let lookup_table_lamports = lookup_table_account.get_lamports();
+        let table_key = *lookup_table_account.get_key();
+        if !lookup_table_account.get_data().is_empty() {
+            ic_msg!(invoke_context, "Table account must not be allocated");
+            return Err(InstructionError::AccountAlreadyInitialized);
+        }
+        drop(lookup_table_account);
+
+        let authority_account =
+            instruction_context.try_borrow_instruction_account(transaction_context, 1)?;
+        let authority_key = *authority_account.get_key();
+        drop(authority_account);
+
+        let payer_account =
+            instruction_context.try_borrow_instruction_account(transaction_context, 2)?;
+        let payer_key = *payer_account.get_key();
+        if !payer_account.is_signer() {
+            ic_msg!(invoke_context, "Payer account must be a signer");
+            return Err(InstructionError::MissingRequiredSignature);
+        }
+        drop(payer_account);
```

Comment:
nit: You could also refactor these checks into a separate function, return the required keys, with a flag to do the signer check